test/e2e/egress_test.go

```diff
@@ -664,6 +666,90 @@ func testEgressMigration(t *testing.T, data *TestData, triggerFunc, revertFunc f
+	if err := NewPodBuilder(fakeExternalName, data.testNamespace, toolboxImage).OnNode(egressNode).WithCommand([]string{"sh", "-c", cmd}).InHostNetwork().Privileged().Create(data); err != nil {
+		t.Fatalf("Failed to create client Pod: %v", err)
+	}
+	defer deletePodWrapper(t, data, data.testNamespace, fakeExternalName)
+	if err := data.podWaitForRunning(defaultTimeout, fakeExternalName, data.testNamespace); err != nil {
+		t.Fatalf("Error when waiting for Pod '%s' to be in the Running state", fakeExternalName)
+	}
+
+	clientPodName := "client-pod"
+	if err := NewPodBuilder(clientPodName, data.testNamespace, toolboxImage).OnNode(egressNode).Create(data); err != nil {
+		t.Fatalf("Error when creating the perftest client Pod: %v", err)
+	}
+	defer deletePodWrapper(t, data, data.testNamespace, clientPodName)
+	if err := data.podWaitForRunning(defaultTimeout, clientPodName, data.testNamespace); err != nil {
+		t.Fatalf("Error when waiting for the perftest client Pod: %v", err)
+	}
+
+	egress := data.createEgress(t, "egress-qos-", nil, map[string]string{"antrea-e2e": clientPodName}, "", egressNodeIP, bandwidth1)
+	_, err := data.waitForEgressRealized(egress)
+	if err != nil {
+		t.Fatalf("Error when waiting for Egress to be realized: %v", err)
+	}
+
+	// expectedBandwidth is Mbps
+	runIperf := func(cmd []string, expectedBandwidth int) {
+		stdout, _, err := data.RunCommandFromPod(data.testNamespace, clientPodName, "toolbox", cmd)
+		if err != nil {
+			t.Fatalf("Error when running iperf3 client: %v", err)
+		}
+		stdout = strings.TrimSpace(stdout)
+		actualBandwidth, _ := strconv.ParseFloat(strings.TrimSpace(stdout), 64)
+		t.Logf("Actual bandwidth: %v Mbits/sec", actualBandwidth)
+		// Allow a certain deviation.
+		assert.InEpsilon(t, actualBandwidth, expectedBandwidth, 0.1)
+	}
+
+	runIperf([]string{"bash", "-c", "iperf3 -c 1.1.1.1 -f m -t 1|grep sender|awk '{print $7}'"}, transMap[bandwidth1.Rate]+transMap[bandwidth1.Burst])
+	runIperf([]string{"bash", "-c", "iperf3 -c 1.1.1.1 -f m -O 1|grep sender|awk '{print $7}'"}, transMap[bandwidth1.Rate])
+
+	toUpdate := egress.DeepCopy()
+	err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		toUpdate.Spec.Bandwidth = bandwidth2
+		_, err := data.crdClient.CrdV1beta1().Egresses().Update(context.TODO(), toUpdate, metav1.UpdateOptions{})
+		if err != nil && errors.IsConflict(err) {
+			toUpdate, _ = data.crdClient.CrdV1beta1().Egresses().Get(context.TODO(), egress.Name, metav1.GetOptions{})
+		}
+		return err
+	})
+	require.NoError(t, err, "Failed to update Egress")
+
+	runIperf([]string{"bash", "-c", "iperf3 -c 1.1.1.1 -f m -t 1|grep sender|awk '{print $7}'"}, transMap[bandwidth2.Rate]+transMap[bandwidth2.Burst])
+	runIperf([]string{"bash", "-c", "iperf3 -c 1.1.1.1 -f m -O 1|grep sender|awk '{print $7}'"}, transMap[bandwidth2.Rate])
+
+	failOnError(data.crdClient.CrdV1beta1().Egresses().Delete(context.TODO(), egress.Name, metav1.DeleteOptions{}), t)
```

Comment:
It should calling it with defer after the Egress is created to ensure it will be deleted regardless of success or failure, to avoid affecting tests.

Reply:
Done.